src/main/java/backend/LoginController.java

```diff
@@ -61,7 +61,8 @@ public void getPreviousLoginDetails() {
 
     public boolean attemptLogin() {
 
-        boolean validRepoId = Utility.isWellFormedRepoId(owner, repo) && !username.isEmpty() && !password.isEmpty();
+        boolean validRepoId = RepoId.isWellFormedRepoIdString(owner + "/" + repo)
+                && !username.isEmpty() && !password.isEmpty();
         boolean loginSuccessful = login(username, password);
 
         if (validRepoId && loginSuccessful) {
```

Comment:
@damithc I'm not sure about this part of the code as I didn't write the LogicController. (I only changed the `Utilty` to `RepoId` because `isWellFormedRepoIdString` was moved from the `Utility` class to the `RepoId` class).

But from what I see it is not checking if the validity repo ID (eg. whether it exists on github). It is really doing it checking if the 4 fields at login is non-empty. I guess I can refactor this to `isFieldsNotEmpty`.